crates/common/src/version.rs

```diff
@@ -0,0 +1,42 @@
+use std::{env, error::Error};
+
+pub const NIGHTLY_VERSION_WARNING_MESSAGE: &str =
+    "This is a nightly build of Foundry. It is recommended to use the latest stable version. \
+    Visit https://book.getfoundry.sh/announcements for more information. \n\
+    To mute this warning set `FOUNDRY_DISABLE_NIGHTLY_WARNING` in your environment. \n";
```

Comment:
We still want to keep nightlies going forward correct? I don't know if this warning pulls its weight

Reply:
We are going to keep doing nightlies but for the majority of users it would be recommended to switch to stable if they are relying on Foundry in a production setting

The plan is to have a 2 week minor stable release cycle accompanied with change logs

Reply:
In the short term the motivation is to have users upgrade their foundryup

We can always update this to a more generic warning to avoid using nightly in production settings, similar to solc

```
Warning (3805): This is a pre-release compiler version, please do not use it in production.
```